plugins/experimental/ssl_session_reuse/src/openssl_utils.cc

```diff
@@ -53,8 +53,12 @@ ssl_new_session(TSSslSessionID &sid)
 
   int session_ret_len = SSL_SESSION_MAX_DER;
   char session_data[SSL_SESSION_MAX_DER];
-  if (!TSSslSessionGetBuffer(&sid, session_data, &session_ret_len)) {
-    TSError("Session data is too large: %d", session_ret_len);
+  const auto buffer_length = TSSslSessionGetBuffer(&sid, session_data, &session_ret_len);
+  if (buffer_length == 0) {
+    TSError("Failed to find a session buffer.");
+    return 0;
+  } else if (buffer_length > session_ret_len) {
+    TSError("Session data is too large. Its size is: %d but our max buffer size is: %d.", buffer_length, SSL_SESSION_MAX_DER);
```

Comment:
Here could you use logic similar to this?
```
    int64_t url_len;
    char url_stack_buf[2048];
    rc = TSHttpHdrEffectiveUrlBufGet(bufp, hdr_loc, url_stack_buf, sizeof(url_stack_buf), &url_len);
    std::unique_ptr<char []> url_heap_buf;
    char *url;
    if (TS_SUCCESS == rc) {
      if (int64_t(sizeof(url_stack_buf)) >= url_len) {
        url = url_stack_buf;
      } else {
        url = new char[url_len];
        url_heap_buf.reset(url);
        rc = TSHttpHdrEffectiveUrlBufGet(bufp, hdr_loc, url, url_len, &url_len);
      }
    }
```

Reply:
This is an interesting idea, but in this cases I don't think we intend to expand the buffer and try again. We're already using `SSL_SESSION_MAX_DER` as the buffer size. 

Before the patch in this PR, any failure with getting the buffer resulted in an error message saying that the data was too large for the buffer. But this isn't always the cause of the problem: it could also be that the buffer lookup fails. This updates the messaging to be more correct.

Thanks for the review.